src/contracts/pods/EigenPod.sol

```diff
@@ -159,143 +113,144 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
+     * change in ACTIVE validator balance is tracked, and any validators with 0 balance are marked `WITHDRAWN`.
+     * @dev Once finalized, the pod owner is awarded shares corresponding to:
+     * - the total change in their ACTIVE validator balances
+     * - any ETH in the pod not already awarded shares
+     * @dev A checkpoint cannot be created if the pod already has an outstanding checkpoint. If
+     * this is the case, the pod owner MUST complete the existing checkpoint before starting a new one.
+     * @param revertIfNoBalance Forces a revert if the pod ETH balance is 0. This allows the pod owner
+     * to prevent accidentally starting a checkpoint that will not increase their shares
      */
-    function verifyBalanceUpdates(
-        uint64 oracleTimestamp,
-        uint40[] calldata validatorIndices,
-        BeaconChainProofs.StateRootProof calldata stateRootProof,
-        bytes[] calldata validatorFieldsProofs,
-        bytes32[][] calldata validatorFields
-    ) external onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_BALANCE_UPDATE) {
-        require(
-            (validatorIndices.length == validatorFieldsProofs.length) && (validatorFieldsProofs.length == validatorFields.length),
-            "EigenPod.verifyBalanceUpdates: validatorIndices and proofs must be same length"
-        );
-
-        // Balance updates should not be "stale" (older than VERIFY_BALANCE_UPDATE_WINDOW_SECONDS)
-        require(
-            oracleTimestamp + VERIFY_BALANCE_UPDATE_WINDOW_SECONDS >= block.timestamp,
-            "EigenPod.verifyBalanceUpdates: specified timestamp is too far in past"
-        );
-
-        // Verify passed-in beaconStateRoot against oracle-provided block root:
-        BeaconChainProofs.verifyStateRootAgainstLatestBlockRoot({
-            latestBlockRoot: eigenPodManager.getBlockRootAtTimestamp(oracleTimestamp),
-            beaconStateRoot: stateRootProof.beaconStateRoot,
-            stateRootProof: stateRootProof.proof
-        });
-
-        int256 sharesDeltaGwei;
-        for (uint256 i = 0; i < validatorIndices.length; i++) {
-            sharesDeltaGwei += _verifyBalanceUpdate(
-                oracleTimestamp,
-                validatorIndices[i],
-                stateRootProof.beaconStateRoot,
-                validatorFieldsProofs[i], // Use validator fields proof because contains the effective balance
-                validatorFields[i]
-            );
+    function startCheckpoint(bool revertIfNoBalance)
+        external
+        onlyEigenPodOwner() 
+        onlyWhenNotPaused(PAUSED_START_CHECKPOINT) 
+    {
+        _startCheckpoint(revertIfNoBalance);
+
+        /// Legacy support for pods deployed pre-M2 that never activated restaking. `startCheckpoint`
+        /// can activate restaking, allowing them to continue using their pods via the checkpoint
+        /// system.
+        if (!hasRestaked) {
+            hasRestaked = true;
+            emit RestakingActivated(podOwner);
```

Comment:
good q. i think no, because `verifyStaleBalance` requires you to have a validator in the `ACTIVE` state -- which should be impossible unless `hasRestaked == true`.

Reply:
(In both M2 and PEPE, `verifyWithdrawalCredentials` requires `hasRestaked == true`)

Reply:
ahh good point! it wouldn't even be possible to trigger `verifyStaleBalance` in this state, so the behavior doesn't matter. that's great!